src/nvim/option.c

```diff
@@ -2030,7 +2030,13 @@ void set_init_1(void)
     } else {
       free(p_enc);
       p_enc = save_enc;
+      /* mb_init() failed. Call it again for the saved encoding */
```

Comment:
// mb_init() on the detected encoding; try again on the default (utf8) encoding.